lib/TPP/ConvertCheckToFunc.cpp

```diff
@@ -0,0 +1,154 @@
+                               std::prev(module.getBody()->end()));
+    func::FuncOp funcOp =
+        rewriter.create<func::FuncOp>(loc, fnName.getValue(), libFnType);
+
+    funcOp->setAttr(LLVM::LLVMDialect::getEmitCWrapperAttrName(),
+                    UnitAttr::get(rewriter.getContext()));
+
+    funcOp.setPrivate();
+  }
+
+  auto call = rewriter.create<func::CallOp>(loc, fnName.getValue(), None,
+                                            op->getOperands());
+  return call;
+}
+
+static SmallVector<Value> getMemRefOperands(OpBuilder &b, Location loc,
+                                            ValueRange operands) {
+  SmallVector<Value> res;
+  res.reserve(operands.size());
+  for (Value op : operands) {
+    auto memrefType = op.getType().dyn_cast<MemRefType>();
+    if (!memrefType)
+      res.push_back(op);
+    else {
+      MemRefType rankedMemref = op.getType().dyn_cast<MemRefType>();
+      UnrankedMemRefType unrankedMemref = UnrankedMemRefType::get(
+          rankedMemref.getElementType(), rankedMemref.getMemorySpace());
+      Value cast = b.create<memref::CastOp>(loc, unrankedMemref, op);
+      res.push_back(cast);
+    }
+  }
+  return res;
+}
+
+static func::CallOp buildExpectAlmostEqualsCall(Location loc, Operation *op,
+                                                ModuleOp module,
+                                                FlatSymbolRefAttr fnName,
+                                                PatternRewriter &rewriter) {
+  auto libFnType = rewriter.getFunctionType(
+      {op->getOperand(0).getType(), op->getOperand(1).getType()}, None);
+
+  if (!module.lookupSymbol(fnName.getAttr())) {
+    OpBuilder::InsertionGuard guard(rewriter);
+    // Insert before module terminator.
+    rewriter.setInsertionPoint(module.getBody(),
+                               std::prev(module.getBody()->end()));
+    func::FuncOp funcOp =
+        rewriter.create<func::FuncOp>(loc, fnName.getValue(), libFnType);
+    funcOp->setAttr(LLVM::LLVMDialect::getEmitCWrapperAttrName(),
+                    UnitAttr::get(rewriter.getContext()));
+    funcOp.setPrivate();
+  }
+
+  func::CallOp call = rewriter.create<func::CallOp>(
+      loc, fnName.getValue(), TypeRange(),
+      getMemRefOperands(rewriter, loc, op->getOperands()));
+  return call;
+}
+
+struct ConvertExpectTrue : public OpRewritePattern<ExpectTrueOp> {
+  ConvertExpectTrue(MLIRContext *context)
+      : OpRewritePattern<ExpectTrueOp>(context, 1) {}
+
+  LogicalResult matchAndRewrite(ExpectTrueOp expectTrueOp,
+                                PatternRewriter &rewriter) const override {
+    Location loc = expectTrueOp.getLoc();
+    std::string kindAsString = "expect_true";
+
+    FlatSymbolRefAttr fnName =
+        SymbolRefAttr::get(rewriter.getContext(), kindAsString);
+
+    ModuleOp module = expectTrueOp->getParentOfType<ModuleOp>();
+
+    buildExpectTrueCall(loc, expectTrueOp, module, fnName, rewriter);
+    expectTrueOp.erase();
```

Comment:
mm, I think you want to notify the rewriter. `rewriter.eraseOp(exepectedTrueOp)`.